sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastNestedLoopJoinExec.scala

```diff
@@ -79,32 +89,46 @@ class GpuBroadcastNestedLoopJoinMeta(
       case GpuBuildRight => right
     }
     verifyBuildSideWasReplaced(buildSide)
+
+    val condition = conditionMeta.map(_.convertToGpu())
+    // Do not yet support AST conditions on anything but semi/anti joins
+    val isAstCondition = join.joinType match {
+      case _: InnerLike => false
+      case LeftSemi | LeftAnti =>
+        conditionMeta.foreach(_.tagForAst())
+        val isAst = conditionMeta.exists(_.canThisBeAst)
+        assert(isAst, s"Non-AST condition in ${join.joinType}")
+        isAst
+      case _ => throw new IllegalStateException(s"${join.joinType} nested loop join not supported")
+    }
+
     val joinExec = ShimLoader.getSparkShims.getGpuBroadcastNestedLoopJoinShim(
       left, right, join,
       join.joinType,
-      None,
+      if (isAstCondition) condition else None,
       conf.gpuTargetBatchSizeBytes)
-    // The GPU does not yet support conditional joins, so conditions are implemented
-    // as a filter after the join when possible.
-    condition.map(c => GpuFilterExec(c.convertToGpu(), joinExec)).getOrElse(joinExec)
+    if (isAstCondition) {
+      joinExec
+    } else {
+      // condition cannot be implemented via AST so fallback to a post-filter if necessary
+      condition.map(c => GpuFilterExec(c, joinExec)).getOrElse(joinExec)
+    }
   }
 }
 
-/**
- * An iterator that does a cross join against a stream of batches.
- */
-class CrossJoinIterator(
+/** Base class for the join iterators based on a nested loop join */
```

Comment:
nit: Is this new with the formatting? Generally I thought the multi-line format was preferred for classes. This is more for my info so I can keep things consistent.

Reply:
It's more my preference.  I prefer to keep one-liner comments as one line, regardless of whether it's a class, variable, method, etc. to save on the vertical whitespace.  I can turn this back into a multi-line comment.

Reply:
I don't care. If the style checker says it is okay then it is okay with me. I just was curious.